hail/src/main/scala/is/hail/expr/ir/InferPType.scala

```diff
@@ -169,7 +244,7 @@ object InferPType {
       case ArrayFlatMap(a, name, body) => {
         InferPType(a, env)
         InferPType(body, env.bind(name, a.pType2.asInstanceOf[PArray].elementType))
-        coerce[PStreamable](a.pType2).copyStreamable(coerce[PIterable](body.pType2).elementType)
+        coerce[PStreamable](a.pType2).copyStreamable(coerce[PIterable](body.pType2).elementType, body.pType2.required)
```

Comment:
I *think* whether or not body.pType2 is required doesn't affect the requiredness of any part of the type of this node---since a `NA` gets flattened the same as an empty array, we don't need to care about the requiredness of `body.pType` to compute this type. The requiredness of the ArrayFlatMap node should be the same as the requiredness of `a.pType2`.